app/models/procedure_presentation.rb

```diff
@@ -79,13 +79,13 @@ def fields
     end
 
     fields.concat(procedure.types_de_champ_for_procedure_presentation
-      .pluck(:type_champ, :libelle, :private, :stable_id)
+      .map { [_1.type_champ, _1.libelle, _1.private, _1.stable_id, _1.revision_type_de_champ.child?] }
       .reject { |(type_champ)| type_champ == TypeDeChamp.type_champs.fetch(:repetition) }
-      .map do |(type_champ, libelle, is_private, stable_id)|
+      .map do |(type_champ, libelle, is_private, stable_id, is_child)|
```

Comment:
plutôt que promener autant d'argument jusqu'à `field_hash_for_type_de_champ_private`, on peut pas juste donner le tdc ?
```suggestion
      .map do |tdc|
```

Reply:
Je ne suis pas convaincu non plus. Avant, on faisait un pluck et pour une bonne raison : il peut y avoir énormément de types de champ ici (issus de toutes les révisions) et un type de champ peut avoir beaucoup de "data" (par exemple : une liste déroulante avec plusieurs milliers d'options). De plus, `type_de_champ.revision_type_de_champ.child?` est sémantiquement faux (mais en pratique, ça marche). J'ai demandé à @E-L-T quelques jours pour y réfléchir si on a une meilleure solution.